serviceLibrary/src/main/java/info/mqtt/android/service/ping/AlarmPingSender.kt

```diff
@@ -21,47 +22,64 @@ import java.util.concurrent.TimeUnit
  *
  * @see MqttPingSender
  */
-internal class AlarmPingSender(val service: MqttService, private val pingLogging: Boolean = false, private val keepPingRecords: Int = 1000) :
-    MqttPingSender {
+internal class AlarmPingSender(val service: MqttService,
+                               val id: String,
+                               private val pingLogging: Boolean = false,
+                               private val keepPingRecords: Int = 1000) : MqttPingSender {
+
 
     private val workManager = WorkManager.getInstance(service)
 
     override fun init(comms: ClientComms) {
-        clientComms = comms
+        clientCommsMap[id] = comms
         messageDatabase = service.messageDatabase
+        Timber.w("Init ping job $id")
     }
 
     override fun start() {
-        // add clientState null check to avoid ClientState.getKeepAlive() NPE(#358,#433)
-        clientComms?.clientState?.let {
-            schedule(clientComms!!.keepAlive)
-        } ?: Timber.e("FIXME: try to start ping schedule, but clientState null, not able to get keepAlive")
+        Timber.d("Start ping job $id")
+            if (clientCommsMap.containsKey(id)) {
```

Comment:
I guess you can remove this line complete because of next line ?
`clientCommsMap[id]?`

Btw, I see the indent is not not right. Please keep an eye on this